Added directory properties

DosenSuppe opened the pull request:
The `FileEditorComponent` can now be set to select directories.

Added two Test-Entries to the demo component.

Preview:
![Aufzeichnung 2025-10-15 151025](https://github.com/user-attachments/assets/8222adbe-3e60-448d-8e94-79270b895d6e)


closes #26

DosenSuppe committed:
Added directory properties

FileEditorComponent can now be set to work with directories.

Added two Test-Entries to the demo component.

closes #26

1nf0rmagician commented on projects/ngx-web-framework/entry-editor/src/file-editor/file-editor.component.html:
```diff
@@ -1,5 +1,5 @@
 <div class="file-editor-container">
-    <input title="File Selector" type="file" class="file-input" (change)="onFileSelected($event)" #fileUpload />
+    <input title="File Selector" type="file" class="file-input" (change)="onFileSelected($event)" [attr.webkitdirectory]="entry().value.unitType === 'Directory' ? '' : null" #fileUpload />
```
Good start here, but please check what is actually uploaded here 😊 The logic of `onFileSelected($event)` only works for a single file atm.

I suggest,  
use `*ngIf` for two separate elements
```html
<input *ngIf="entry().value.unitType === 'Directory'"
       type="file"
       class="file-input"
       (change)="onDirectorySelected($event)"
       webkitdirectory
       #fileUpload />
<input *ngIf="entry().value.unitType !== 'Directory'"
       type="file"
       class="file-input"
       (change)="onFileSelected($event)"
       #fileUpload />
```

refactor the `onFileSelected` method and add a `OnDirectorySelected` method

1nf0rmagician commented:
FYI: @H4llJ0rdan

DosenSuppe committed:
Address review comment

- Added input element for both File- and Directory-upload.
- The upload process is handled through the new `uploadFiles()` method
- Added `onDirectorySelected()` method and refactored `onFileSelected()`

DosenSuppe commented:
The `EntryValue` interface is not suitable for uploading a collection of files.

Previously `EntryValue.currentValue` was set one file's content when contacting the endpoint.
For a single file that works fine but fails when handling directories; commonly having a collection of files.
